reconcile/templating/renderer.py

```diff
@@ -90,8 +88,34 @@ def read(self, path: str) -> Optional[str]:
         return None
 
 
-class GitlabFilePersistence(FilePersistence):
-    def __init__(self, vcs: VCS, mr_manager: MergeRequestManager) -> None:
+class PersistenceContext(FilePersistence):
```

Comment:
same as previous pr, use something like `transaction` or docstring to indicate this will persistent at the end.